packages/cel/src/object.ts

```diff
@@ -0,0 +1,400 @@
+      if (!isCelMap(v)) {
+        throw unexpectedTypeError(field, "map", v);
+      }
+      const map = reflectMap(field, undefined, false);
+      copyMap(map, v);
+      msg.set(field, map);
+      return;
+    case "enum":
+      msg.set(field, enumFromCel(field, v as CelValue));
+      return;
+    case "message":
+      msg.set(field, msgFromCel(field, v));
+      return;
+  }
+  msg.set(field, scalarFromCel(field, field.scalar, v));
+}
+
+function copyList(list: ReflectList, cList: CelList) {
+  const field = list.field();
+  for (const v of cList) {
+    switch (field.listKind) {
+      case "enum":
+        list.add(enumFromCel(field, v as CelValue));
+        break;
+      case "message":
+        list.add(msgFromCel(field, v as CelValue));
+        break;
+      case "scalar":
+        list.add(scalarFromCel(field, field.scalar, v as CelValue));
+        break;
+    }
+  }
+}
+
+function copyMap(map: ReflectMap, cMap: CelMap) {
+  const field = map.field();
+  for (const [k, v] of cMap.entries()) {
+    const key = scalarFromCel(field, field.mapKey, k);
+    switch (field.mapKind) {
+      case "enum":
+        map.set(key, enumFromCel(field, v as CelValue));
+        break;
+      case "scalar":
+        map.set(key, scalarFromCel(field, field.scalar, v as CelValue));
+        break;
+      case "message":
+        map.set(key, msgFromCel(field, v as CelValue));
+        break;
+    }
+  }
+}
+
+function enumFromCel(field: DescField, v: CelValue) {
+  if (typeof v !== "bigint") {
+    throw unexpectedTypeError(field, "int", v);
+  }
+  return Number(v);
+}
+
+function msgFromCel(
+  field: DescField & { message: DescMessage },
+  v: CelValue,
+): ReflectMessage {
+  if (isWrapperDesc(field.message)) {
+    const msg = reflect(field.message, undefined, false);
+    msg.set(
+      field.message.fields[0],
+      scalarFromCel(field.message.fields[0], field.message.fields[0].scalar, v),
+    );
+    return msg;
+  }
+  switch (field.message.typeName) {
+    case AnySchema.typeName:
+      if (isReflectMessage(v, AnySchema)) {
+        return v;
+      }
+      return reflect(AnySchema, anyFromCel(v));
+    case ValueSchema.typeName:
+      return reflect(ValueSchema, valueFromCel(v));
+    case StructSchema.typeName:
+      if (!isCelMap(v)) {
+        throw unexpectedTypeError(field, "map", v);
+      }
+      return reflect(StructSchema, structFromCel(v));
+    case ListValueSchema.typeName:
+      if (!isCelList(v)) {
+        throw unexpectedTypeError(field, "list", v);
+      }
+      return reflect(ListValueSchema, listValueFromCel(v));
+  }
+  if (!isReflectMessage(v, field.message)) {
+    throw unexpectedTypeError(field, field.message.typeName, v);
+  }
+  return v;
+}
+
+/**
+ * Converts a CelValue to google.protobuf.Any.
+ *
+ * While the CEL spec doesn't say anything about converting
+ * CEL values to types, there a couple conformance tests around
+ * WKTs that expect this behavior. The go implementation also handles
```

Comment:
The go code is spread across multiple files. Every value, except for type, implements a conversion into Any in go.